### ISSUE TITLE
crash in macOS when forking

### ISSUE THREAD

**Issue by Jean-Daniel** [kapicorp/kapitan#704]
**Describe the bug/feature**
Most of macOS frameworks and runtime libraries are not fork safe. While using only low-level libraries may work, in practice, even Python relies on some fork unsafe code, which make it globally fork unsafe as can be shown by this crash when trying to fetch a HTTP file.

```
Dependency https://github.com/…/controller.yaml: fetching now
objc[27302]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
objc[27302]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
```

**To Reproduce**
1. try to fetch/compile a simple inventory containing a HTTP dependency on macOS 11/python 3.9 (maybe crash with other versions too).

See above.

**Expected behavior**
Do not crash.

** If it's a bug (please complete the following information):**
 - `python --version`: Python 3.9.2 (brew)
 - `Are you using pyenv or virtualenv?`: python3 -m venv

**Additional context**
macOS 11.2.2

**Comment by Jean-Daniel** [comment]
For the record:

https://docs.python.org/3/library/multiprocessing.html

> Changed in version 3.8: On macOS, the spawn start method is now the default. The fork start method should be considered unsafe as it can lead to crashes of the subprocess. See [bpo-33725](https://bugs.python.org/issue33725).

**Comment by ramaro** [comment]
Thanks for reporting @Jean-Daniel ! It's unfortunately a very frequent problem. I guess going forward, we should remove multiprocessing from Kapitan when go-jsonnet is good and async http fetching instead for dependencies.
This also reminds me that we need to setup a macOS pipeline to run tests as we currently only do that on Linux.

**Comment by MatteoVoges** [comment]
resolved by #753
